cob_twist_controller/src/limiters/limiter.cpp

```diff
@@ -320,3 +356,46 @@ KDL::JntArray LimiterIndividualJointAccelerations::enforceLimits(const KDL::JntA
     return q_dot_norm;
 }
 /* END LimiterIndividualJointAccelerations **********************************************************************/
+
+/* BEGIN LimiterCartesianVelocities ********************************************************************/
+/**
+ * This implementation implements a saturation function to the Cartesian twists
+ */
+KDL::Twist LimiterCartesianVelocities::enforceLimits(const KDL::Twist& v_in) const
+{
+    KDL::Twist v_out(v_in);
+    // linear limts
+
+    if (v_in.rot.x() >limiter_params_.max_rot_twist)
+        v_out.rot.x(limiter_params_.max_rot_twist);
+    if (v_in.rot.x() <-limiter_params_.max_rot_twist)
+        v_out.rot.x(-limiter_params_.max_rot_twist);
+
+    if (v_in.rot.y() >limiter_params_.max_rot_twist)
+        v_out.rot.y(limiter_params_.max_rot_twist);
+    if (v_in.rot.y() <-limiter_params_.max_rot_twist)
+        v_out.rot.y(-limiter_params_.max_rot_twist);
+
+    if (v_in.rot.z() >limiter_params_.max_rot_twist)
+        v_out.rot.z(limiter_params_.max_rot_twist);
+    if (v_in.rot.z() <-limiter_params_.max_rot_twist)
+        v_out.rot.z(-limiter_params_.max_rot_twist);
+
+    if (v_in.vel.x() >limiter_params_.max_lin_twist)
+        v_out.vel.x(limiter_params_.max_lin_twist);
+    if (v_in.vel.x() <-limiter_params_.max_lin_twist)
+        v_out.vel.x(-limiter_params_.max_lin_twist);
+
+    if (v_in.vel.y() >limiter_params_.max_lin_twist)
+        v_out.vel.y(limiter_params_.max_lin_twist);
+    if (v_in.vel.y() <-limiter_params_.max_lin_twist)
+        v_out.vel.y(-limiter_params_.max_lin_twist);
+
+    if (v_in.vel.z() >limiter_params_.max_lin_twist)
+        v_out.vel.z(limiter_params_.max_lin_twist);
+    if (v_in.vel.z() <-limiter_params_.max_lin_twist)
+        v_out.vel.z(-limiter_params_.max_lin_twist);
```

Comment:
this implementation might change the "direction" of the twist, as it is cropping rather than scaling down the length of the vector while keeping the direction...
it is analogue to the `LimiterIndividualJoint*` implementations...
I think we should also have a case where if one linear direction (x,y,z) exceeds the limits the whole linear velocity is to be scaled by the same factor...